chips/sam4l/src/pm.rs

```diff
@@ -259,243 +148,170 @@ pub enum SystemClockSource {
     /// Use the internal digital frequency locked loop (DFLL) sourced from
```

Comment:
Yeah, that was very much on purpose. While I agree that it would be good to have a general solution, it's a lot of code that would currently go utterly unused. The setup we've had for a while now only actually worked with an external oscillator at 16 MHz and with a system clock of 48 MHz, but it wasn't clear that there were no choices.

In the future, the idea was that if someone really wants to use say at 12 MHz crystal, they would add it as an option to the list of enums, and add functionality to support it.